File: Google Codejam/2013/Lawnmower/main.cpp
#include <iostream>
#include <fstream>
#include <vector>

using namespace std;

int main() {
    int T; cin >> T;
    for (int i = 1; i <= T; ++i) {
        int N, M; cin >> N >> M;

        vector< vector<int> > A(N, vector<int>(M));
        for (int i = 0; i < N; ++i)
            for (int j = 0; j < M; ++j)
                cin >> A[i][j];

        bool ok = true;
        while (ok) {
            int minim = 500;
            int x = -1, y = -1;

            for (int i = 0; i < N; ++i)
                for (int j = 0; j < M; ++j)
                    if (A[i][j] != -1 && A[i][j] < minim) {
                        minim = A[i][j];
                        x = i; y = j;
                    }

            if (minim == 500)
                break;

            bool row = true, column = true;
            for (int i = 0; i < N; ++i)
                if (A[i][y] != -1 && A[i][y] != minim) {
                    column = false;
                    break;
                }
            
            for (int j = 0; j < M; ++j)
                if (A[x][j] != -1 && A[x][j] != minim) {
                    row = false;
                    break;
                }

            if (!row and !column) {
                ok = false;
                break;
            }

            if (row) {
                for (int j = 0; j < M; ++j)
                    A[x][j] = -1;
            } else {
                for (int i = 0; i < N; ++i)
                    A[i][y] = -1;
            }
        }

        cout << "Case #" << i << ": ";
        if (ok)
            cout << "YES\n";
        else
            cout << "NO\n";
    }
}

File: Google Codejam/2013/Lawnmower/main.py
def read():
    return [int(x) for x in input().split(' ')]

T = int(input())

for test in range(1, T + 1):
    N, M = read()
    A = [read() for i in range(N)]

    ok = True
    while ok == True:
        minim = 500
        where = -1, -1
        for i in range(N):
            for j in range(M):
                if A[i][j] == -1 and A[i][j] < minim:
                    minim = A[i][j]
                    where = i, j

        if minim == 500:
            break

        # check whether it's the column or the row
        x, y = where
        column = True
        row = True
        for i in range(N):
            if A[i][y] != -1 and A[i][y] != minim:
                column = False
                break

        for j in range(M):
            if A[x][j] != -1 and A[x][j] != minim:
                row = False
                break

        if column == False and row == False:
            ok = False
            break

        if row == True:
            for j in range(M):
                A[x][j] = -1
        else:
            for i in range(N):


